c7n/mu.py

```diff
@@ -119,18 +120,27 @@ def create(self):
             arc_prefix = os.path.relpath(root, venv_lib_path)
             files = self.filter_files(files)
             for f in files:
-                self._zip_file.write(
-                    os.path.join(root, f),
-                    os.path.join(arc_prefix, f))
+                f_path = os.path.join(root, f)
+                dest_path = os.path.join(arc_prefix, f)
+                self.add_file(f_path, dest_path)
 
     def add_file(self, src, dest):
-        self._zip_file.write(src, dest)
+        info = zipfile.ZipInfo(dest)
+        timestamp = os.path.getmtime(src)
+        info.date_time = datetime.timetuple(
+            datetime.fromtimestamp(timestamp)
+        )
+        fp = open(src, 'rb')
+        contents = fp.read()
+        fp.close()
+        self.add_contents(info, contents)
 
     def add_contents(self, dest, contents):
         if not isinstance(dest, zipfile.ZipInfo):
             dest = zinfo(dest)
         # see zinfo function for some caveats
         assert not self._closed, "Archive closed"
+        dest.external_attr = 0444 << 16L
```

Comment:
this makes the line in the zinfo method redundant as well. it would be good to have a test verifying this is behaving as intended.